src/sparseml/exporters/transforms/conv_to_qlinearconv.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy
+from onnx import ModelProto, helper, numpy_helper
+
+from sparseml.exporters.transforms.onnx_transform import OnnxTransform
+from sparseml.exporters.transforms.utils.helpers import (
+    attribute_to_kwarg,
+    delete_quant_node,
+    get_quantization_params,
+    quantize_array,
+)
+from sparseml.exporters.transforms.utils.matching import (
+    INITIALIZER_MATCH,
+    MatchResult,
+    get_structural_matches,
+)
+from sparseml.onnx.utils.graph_editor import (
+    ONNXGraph,
+    remove_node_and_params_from_graph,
+)
+from sparseml.onnx.utils.helpers import get_init_by_name, get_node_output_nodes
+
+
+class ConvToQLinearConv(OnnxTransform):
+    """
+    Transforms
+
+    ```
+                        weight (initializer)
+                        |
+    input               QuantizeLinear
+    |                   |
+    DequantizeLinear    DequantizeLinear
+    |                   |
+            Conv
+            |
+            QuantizeLinear
+    ```
+
+    into
+
+    `input -> QLinearConv`
+    """
+
+    def transform(self, model: ModelProto) -> ModelProto:
+        graph = ONNXGraph(model)
+        matches = get_structural_matches(
+            graph,
+            op_type="Conv",
+            parent_ops=[
+                ["DequantizeLinear"],
+                [INITIALIZER_MATCH, "QuantizeLinear", "DequantizeLinear"],
+            ],
+            children_ops=[["QuantizeLinear"]],
+        )
+        for match in matches:
+            self._do_transform(model, match)
+        ONNXGraph(model).sort_nodes_topologically()
+        return model
+
+    def _do_transform(self, model: ModelProto, match: MatchResult):
```

Comment:
Want to stick to my convention and have those functions outside the class? No reason for it, other than readability (however, this is subjective) and consistency across all the passes -> for the reader's comfort.

Reply:
Also, this function should be public and have some docstring with explanation

Reply:
I think having interior private function makes more sense - we are exporting the Transform object, not the singular function for doing one transform operation. Since the match object that is passed in is tightly coupled to the match result, you shouldn't really use it separately from that IMO. Unless we move the transform to actually take in the nodes like it was done before, then having a public function exterior would be fine.

Reply:
agree with your logic @corey-nm. to the point of standardizing to moving things out - it could help with variable scoping, so things don't leak